File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n=prices.size();
        int buy_day=prices[0];
        int sale_day=prices[0];
        int max_sale=0;
        int x;
        for(int i=1;i<n;i++){
            if(buy_day>prices[i]){
                buy_day=prices[i];
            }
            else {
                max_sale=max(max_sale,prices[i]-buy_day);

            }
            
        }
        

        return max_sale;
        
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n=prices.length;
        int profit=0;
        int maxprofit=0;
        int buy=prices[0];
        for(int i=1;i<n;i++){
            int sell=prices[i];
            if(sell<buy){
                buy=sell;
            }
            else{
            profit=sell-buy;
            if(profit>maxprofit){
                maxprofit=Math.max(profit,maxprofit);
            }
            }
        }
        return maxprofit;
        
    }
}